### PR TITLE
Branch update dg

### PR DESCRIPTION
**Pull request by jadenlimjc**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -38,6 +43,17 @@ The architecture of EasInternship is designed to follow the MVC (Model-View-Cont
 - **Model (InternshipList)**: Manages the state of the application, including the list of internships and tasks.
 - **Storage**: Responsible for loading and saving data from and to the disk.
 
+#### Overview
+
+The `EasInternship` class serves as the entry point of the application. It manages the application's main loop, where the user is continually prompted for input, and commands are processed in response. The class is responsible for initializing the UI, loading saved data, and handling user input until the user chooses to exit the program.
+
+### Key Responsibilities
```
Comment (Ridiculouswifi) [inline docs/DeveloperGuide.md:50]: Perhaps one more "#" as it is on the same level as "Overview"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -326,10 +332,23 @@ The `RemoveCommand` class is responsible for removing the values within fields o
 The following sequence diagram shows how the `RemoveCommand` is executed:
 ![](UML/RemoveCommand_Sequence_Diagram.png)
 
-### Product scope
+## Product scope
 ### Target user profile
```
Comment (Ridiculouswifi) [inline docs/DeveloperGuide.md:336]: Do we have an index page for sections from this section onwards?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -364,12 +383,71 @@ This product helps users streamline the process of tracking and managing their i
...
+7. Scalability: The application should handle an extensive number of internship entries (e.g., 100+ entries) without a noticeable decrease in performance.
+8. Data Persistence: All data should be automatically saved to a file, allowing users to resume their progress upon reopening the application.
+
 
 ## Glossary
 
-* *glossary item* - Definition
+* *Internship* - The position of a student or trainee who works in an organisation, with associated role, company, duration and deadline information.
+*  *InternshipList* - A collection of Internships added by the user.
+*  *Deadline* - The date which an event is schedule for.
```
Comment (Ridiculouswifi) [inline docs/DeveloperGuide.md:400]: I think the word "schedule" should be in past tense.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -364,12 +383,71 @@ This product helps users streamline the process of tracking and managing their i
...
+*  *Parser* - The component responsible for interpreting user commands and executing the appropriate actions.
+*  *Storage* - The component responsible for saving and loading the user's data to and from a file. 
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+1. **Installation and Setup:**
+   - Ensure Java Runtime Environment (JRE) is installed on your system.
```
Comment (Ridiculouswifi) [inline docs/DeveloperGuide.md:407]: Should we specify java 17?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -364,12 +383,71 @@ This product helps users streamline the process of tracking and managing their i
...
+   - Test the `exit` command to ensure the application exits properly.
+
+
+4. **Adding an Internship:**
+   - Enter the command `add -role {role name} -company {company name}` eg. `add -role Software Engineer Intern -company Google`
+   - Verify that the internship has been added successfully.
+
+   
+5. **Updating an Internship:**
+  - Enter the command `update {ID} {-field} {updated information}` eg. `update 1 -skills python, java`.
```
Comment (Ridiculouswifi) [inline docs/DeveloperGuide.md:428]: Under the updated information, even though it is valid to use the comma in such a way, can we not put it as an example. I think having to use commas is quite confusing for a user

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -364,12 +383,71 @@ This product helps users streamline the process of tracking and managing their i
...
+6. **Adding an Internship to favourites:**
+    - Enter the command `mark {ID}` eg 
+
+
+7. **Displaying InternshipList**
+   - Enter the command `list` to show the most updated `InternshipList`.
+
+
+8. **Sorting the InternshipList**
+   - Enter the command `sort {field}` eg. `sort deadline`.
```
Comment (Ridiculouswifi) [inline docs/DeveloperGuide.md:441]: The field should have a dash in front of it

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -364,12 +383,71 @@ This product helps users streamline the process of tracking and managing their i
...
+   - Verify that the list displayed is sorted according to the entered field.
+
+
+9. **Filtering the InternshipList**
+   - Enter the command `filter {field} {field data}` eg. `filter -company Google`.
```
Comment (Ridiculouswifi) [inline docs/DeveloperGuide.md:446]: The {field} portion should also have a dash in front of it

**Review by Ridiculouswifi — APPROVED** [review]
Good updates Jaden!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (jadenlimjc) [reply docs/DeveloperGuide.md:336]: Hi Ting Hui, what do you mean by index page? If you are referring to the section header above, I have added the hyperlink references above.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (jadenlimjc) [reply docs/DeveloperGuide.md:428]: Understood! Removed.

**Pull request merged**